inc/item_softwarelicense.class.php

```diff
@@ -0,0 +1,987 @@
+         echo "</td></tr>";
+
+         echo "</table>";
+         Html::closeForm();
+         $ajax_url = $CFG_GLPI['root_doc'] . '/ajax/dropdownAllItems.php';
+         $js = <<<JAVASCRIPT
+function updateItemDropdown(itemtype_el) {
+   $.ajax({
+      method: "POST",
+      url: "$ajax_url",
+      data: {
+         name: 'items_id',
+         idtable: itemtype_el.value
+      },
+      success: function(data) {
+         $("[name='items_id']").select2('destroy').empty().replaceWith(data);
+      }
+   });
+}
+JAVASCRIPT;
+         echo Html::scriptBlock($js);
+      }
+
+      if ($number < 1) {
+         echo "<table class='tab_cadre_fixe'>";
+         echo "<tr><th>".__('No item found')."</th></tr>";
+         echo "</table></div>\n";
+         return;
+      }
+
+      // Display the pager
+      Html::printAjaxPager(__('Affected items'), $start, $number);
+
+      $queries = [];
+      foreach ($CFG_GLPI['software_types'] as $itemtype) {
+         $canshowitems[$itemtype] = $itemtype::canView();
+         $itemtable = $itemtype::getTable();
+         $query = [
+            'SELECT' => [
+               $item_license_table . '.*',
+               'glpi_softwarelicenses.name AS license',
+               'glpi_softwarelicenses.id AS vID',
+               'glpi_softwarelicenses.softwares_id AS softid',
+               "{$itemtable}.name AS itemname",
+               "{$itemtable}.id AS iID",
+               new QueryExpression("'{$itemtype}' AS {$DB::quoteName('item_type')}"),
+            ],
+            'FROM'   => $item_license_table,
+            'INNER JOIN' => [
+               'glpi_softwarelicenses' => [
+                  'FKEY'   => [
+                     $item_license_table     => 'softwarelicenses_id',
+                     'glpi_softwarelicenses' => 'id'
+                  ]
+               ]
+            ],
+            'LEFT JOIN' => [
+               $itemtable => [
+                  'FKEY'   => [
+                     $item_license_table     => 'items_id',
+                     $itemtable        => 'id', [
+                        'AND' => [
+                           $item_license_table.'.itemtype'  => $itemtype
+                        ]
+                     ]
+                  ]
+               ]
+            ],
+            'WHERE'     => [
+               'glpi_softwarelicenses.id'                   => $searchID,
+               'glpi_items_softwarelicenses.is_deleted'     => 0
+            ]
+         ];
+         if ($DB->fieldExists($itemtable, 'serial')) {
+            $query['SELECT'][] = $itemtable.'.serial';
+         } else {
+            $query['SELECT'][] = "'' AS ".$itemtable.".serial";
+         }
+         if ($DB->fieldExists($itemtable, 'otherserial')) {
+            $query['SELECT'][] = $itemtable.'.otherserial';
+         } else {
+            $query['SELECT'][] = "'' AS ".$itemtable.".otherserial";
```

Comment:
```suggestion
            $query['SELECT'][] = new QueryExpression($DB->quoteValue('') . " AS " . $DB->quoteName($itemtable.".otherserial"));
```
also for other uses of `value as field` in this PR.